refs/marshal.go

```diff
@@ -122,16 +115,14 @@ func (o *ObjectID) StableSize() int {
 
 // ObjectIDNestedListMarshal writes protobuf repeated ObjectID field
 // with fNum number to buf.
-func ObjectIDNestedListMarshal(fNum int64, buf []byte, ids []ObjectID) (off int, err error) {
+func ObjectIDNestedListMarshal(fNum int64, buf []byte, ids []ObjectID) (off int) {
+	prefix, _ := proto.NestedStructurePrefix(fNum)
 	for i := range ids {
-		var n int
+		off += binary.PutUvarint(buf, prefix)
 
-		n, err = proto.NestedStructureMarshal(fNum, buf[off:], &ids[i])
-		if err != nil {
-			return
-		}
-
-		off += n
+		n := ids[i].StableSize()
+		off += binary.PutUvarint(buf[off:], uint64(n))
+		off += proto.BytesMarshal(objectIDValField, buf, ids[i].val)
```

Comment:
Why not `proto.NestedStructureMarshal` there?

Reply:
The current implementation is faster.: we don't compare definitely-non-nil. don't perform varint-encoding of `fNum` multiple times and don't perform runtime dispatch of `StableMarshal`.

Reply:
So this optimization makes sense for all slices of nested structures. I guess there are more than one case for it. 

Have you considered having this code in `proto` package or it is heavily depended on the context and cannot be generalized in separate function?

Reply:
It cannot be generalized, because the implementation (`BytesMarshal`) uses knowledge about the type (`ObjectID`). The main problem with generalization for `NestedStructureMarshal` is the lack of a proper signature for `[]Type` where `Type` is generic.
We can apply similar optimization for other lists too (we have about 3-4 of them), this list caught my eye first.